packages/flame/lib/src/nine_tile_box.dart

```diff
@@ -41,20 +44,114 @@ class NineTileBox {
       : tileSize = tileSize ?? sprite.src.width ~/ 3 {
     this.destTileSize = destTileSize ?? this.tileSize;
     final centerEdge = this.tileSize.toDouble();
-    _center = Rect.fromLTWH(centerEdge, centerEdge, centerEdge, centerEdge);
+    center = Rect.fromLTWH(centerEdge, centerEdge, centerEdge, centerEdge);
     _dst = Rect.fromLTWH(0, 0, this.destTileSize * 3, this.destTileSize * 3);
   }
 
+  /// Creates a nine-box instance with the specified grid size
+  ///
+  /// The outside columns and rows can be specified with fixed width or height.
+  /// The center column and row take up the remaining space.
+  NineTileBox.withGrid(
+    this.sprite, {
+    double leftColumnWidth = 0.0,
+    double rightColumnWidth = 0.0,
+    double topRowHeight = 0.0,
+    double bottomRowHeight = 0.0,
```

Comment:
What do you think about dropping `Row` and `Column` from the names? Would it be less clear what they mean?

Reply:
Initially I had just `left, right, top, bottom` but that was confusing as the values here don't work the same as a regular Rect - but I think `_Width, _Height` will be ok. Will make the change.

Reply:
I think the top one is slightly clearer, especially since we usually use size described by a Vector2 that represents both width and height.

Reply:
would `border` work? columns and rows made me a little confused here.

Reply:
So `leftBorderWidth` etc? Could work, hard do come up with something super clear here.

Reply:
Personally, border is confusing to me because it suggests it is outside (or surrounds) the NineTileBox. I prefer leftWidth, topHeight etc

Reply:
Let's go with that then, I think those are fine.
Can you add them to the dartdocs too, explaining what each of them do and it'll be even clearer?
You can refer to arguments in the dartdoc with `[leftBorderWidth]` etc.